src/presign/participant.rs

```diff
@@ -489,42 +560,44 @@ impl PresignParticipant {
         info!("Presign: Handling round one message.");
 
         // First check that we have the round one public broadcast from this
-        // participant. If not, we cannot proceed, so stash that message and
-        // continue.
-        if !self
+        // participant. If not, we cannot proceed, so stash that message.
+        let r1_public_broadcast = match self
             .local_storage
-            .contains::<storage::RoundOnePublicBroadcast>(message.from())
+            .retrieve::<storage::RoundOnePublicBroadcast>(message.from())
         {
-            // Store any early round one messages.
-            info!("Presign: Stashing early round one message.");
-            self.stash_message(message)?;
-            return Ok(ProcessOutcome::Incomplete);
-        }
-        let r1_public_broadcast = self
-            .local_storage
-            .retrieve::<storage::RoundOnePublicBroadcast>(message.from())?;
+            Ok(m) => m,
+            Err(_) => {
+                // XXX This error may either be due to (1) an item not being
+                // found in storage, or (2) an internal issue with storage.
+                // Currently, this code path _assumes_ that the reason this
+                // error occurs is due to (1)!
```

Comment:
suggestion: Ooh, good catch. Can this be fixed by doing a conditional to check if `local_storage.contains` the message, and if so `retrieve`-ing it?

Reply:
Ooh, `StorageItemNotFound` is being refactored out in #332! So that is not a reliable way to figure out the difference here.

Reply:
Oh yes, good point! Will fix!